modules/tensor_mechanics/src/materials/ComputeFiniteStrain.C

```diff
@@ -174,3 +128,91 @@ ComputeFiniteStrain::computeQpStrain()
   _mechanical_strain[_qp] = _rotation_increment[_qp] * _mechanical_strain[_qp] * _rotation_increment[_qp].transpose();
   _total_strain[_qp] = _rotation_increment[_qp] * _total_strain[_qp] * _rotation_increment[_qp].transpose();
 }
+
+void
+ComputeFiniteStrain::computeQpIncrements(RankTwoTensor & total_strain_increment, RankTwoTensor & rotation_increment)
+{
+  if (_decomposition_method == "Rashid")
```

Comment:
Take a look at the implementation in NumDOFs. You'll see a class scoped enum in the header that's populated with `MooseEnum::getEnum()`. Then you can do much more efficient switching or "if" statements.

https://github.com/idaholab/moose/blob/devel/framework/src/postprocessors/NumDOFs.C

One thing to note, please use the new C++11 enum declaration instead of the one here.

```
enum class Foo
{
   ...
};
```

Then you have to use the namespace of the enum when you use it in your .C file.